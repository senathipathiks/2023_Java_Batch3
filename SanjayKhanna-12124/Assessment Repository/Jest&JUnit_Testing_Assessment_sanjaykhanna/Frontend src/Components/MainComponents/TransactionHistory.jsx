import React, { useEffect, useState } from 'react';

const services = require('../Services/Services')

const TransactionHistory = () => {
    const [data, setData] = useState([]);
    var count = 1;

    useEffect(() => {
        services.doGetAllTransaction().then((res) => {
            return res.data;
        }).then((resp) => {
            setData(resp)
        }).catch(err => {
            console.log(err)
        })
    }, [])

    return (
        <div>
            <section className="container container-lg">
                <div className="card card-lg mt-5 shadow-lg p-2 mx-auto">
                    <div className="card-header card-title border-0 text-center h1">History Of Transaction</div>
                    <div className="card-body">
                        <div className="container row mt-3">
                            <div className="col-6" style={{ "textAlign": "left" }}>
                            </div>
                        </div>
                        <div className='container'>
                            <table data-testid='historyTable' className="table table-bordered table-striped table-hover mt-3">
                                <thead className="bg-secondary text-white text-center">
                                    <tr>
                                        <td>Transaction Number</td>
                                        <td>Transaction Date<small style={{ fontSize: '12px', marginLeft: '5px' }}>(DD/MM/YYYY)</small></td>
                                        <td>Transaction Time<small style={{ fontSize: '12px', marginLeft: '5px' }}>(hh/mm/ss)</small></td>
                                        <td>Transaction Amount</td>
                                        <td>Account Number</td>
                                        <td>Account Holder</td>
                                        <td>Transaction Status</td>
                                    </tr>
                                </thead>
                                <tbody>
                                    {data &&
                                        data.map((item) => (
                                            <tr key={item.transactionId}>
                                                <td>{count++}</td>
                                                <td>{new Date(item.transactionDate).getDate()}/{new Date(item.transactionDate).getMonth() + 1}/{new Date(item.transactionDate).getFullYear()}</td>
                                                <td>{new Date(item.transactionDate).getHours()}:{new Date(item.transactionDate).getMinutes()}:{new Date(item.transactionDate).getSeconds()}</td>
                                                <td>{item.amount}</td>
                                                <td>{item.payee.account.accountNumber}</td>
                                                <td>{item.payee.payeeName}</td>
                                                <td>{item.status}</td>
                                            </tr>
                                        ))
                                    }
                                </tbody>
                            </table>
                        </div>
                    </div>
                </div>
            </section>
        </div>
    );
}

export default TransactionHistory;
