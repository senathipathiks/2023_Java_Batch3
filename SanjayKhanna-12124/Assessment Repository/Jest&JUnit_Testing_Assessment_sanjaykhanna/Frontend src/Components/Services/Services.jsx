
import axios from 'axios';
const origin = "http://localhost:1330";
const addPayee = "/addPayee"
const deletePayee = "/deletePayee?payeeId="
const updatePayee = "/updatePayee"
const getPayees = "/getPayees"
const getPayee = "/getPayee?payeeId="
const saveTransaction = "/saveTransaction"
const getTransactions = "/getTransactions"

const doAdd = (data) => {
    return axios.post(origin + addPayee, data);
}

const doDelete = (payeeId) => {
    return axios.delete(origin + deletePayee + payeeId);
}

const doUpdate = (data) => {
    return axios.put(origin + updatePayee, data);
}

const doGetPayee = (payeeId) => {
    return axios.get(origin + getPayee + payeeId);
}

const doGetAllPayee = () => {
    return axios.get(origin + getPayees);
}

const doSave = (data) => {
    return axios.post(origin + saveTransaction, data);
}

const doGetAllTransaction = () => {
    return axios.get(origin + getTransactions);
}

// exports.doAdd = doAdd;
// exports.doDelete = doDelete;
// exports.doGetAllPayee = doGetAllPayee;
// exports.doGetAllTransaction = doGetAllTransaction;
// exports.doGetPayee = doGetPayee;
// exports.doSave = doSave;
// exports.doUpdate = doUpdate;


export { doAdd, doDelete, doGetAllPayee, doGetAllTransaction, doGetPayee, doSave, doUpdate };

