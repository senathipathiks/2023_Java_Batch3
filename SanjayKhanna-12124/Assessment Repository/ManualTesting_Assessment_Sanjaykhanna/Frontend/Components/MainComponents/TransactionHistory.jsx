import React, { useEffect, useState } from 'react';
import transactionService from '../../Services/TransactionService';
import Search from '../UtilitiesComponents/Search';

const TransactionHistory = () => {
    const [data, setData] = useState([]);
    const [search, setSearch] = useState("");
    const [count,setCount] = useState(1)

    useEffect(() => {
        transactionService.doGetAll()
            .then((res) => {
                setData(res.data);
            })
            .catch((err) => {
                console.log(err);
            })
    },[])

    return (
        <div>
            <section className="container container-lg">
                <div className="card card-lg mt-5 shadow-lg p-2 mx-auto">
                    <div className="card-header card-title border-0 text-center h1">History Of Transaction</div>
                    <div className="card-body">
                    <div className="container row mt-3">
                            <div className="col-6" style={{ "textAlign": "left" }}>
                                </div>
                            <div className="col-6">
                                <Search search={search} setSearch={setSearch} />
                            </div>
                            </div>
                        <div className='container'>
                            <table className="table table-bordered table-striped table-hover mt-3">
                                <thead className="bg-dark text-white text-center">
                                    <tr>
                                        <td>Transaction Number</td>
                                        <td>Transaction Date</td>
                                        <td>Transaction Amount</td>
                                        <td>Account Number</td>
                                        <td>Account Holder</td>
                                        <td>Transaction Status</td>
                                    </tr>
                                </thead>
                                <tbody>
                                    {
                                        data.filter((st) => st.payee.payeeName.toLowerCase().includes(search) || st.payee.payeeName.toUpperCase().includes(Search)).map((item) => (
                                            <tr key={item.transactionId}>
                                                <td>{count && setCount(count+1)}</td>
                                                <td>{item.transactionDate}</td>
                                                <td>{item.amount}</td>
                                                <td>{item.payee.accountNumber}</td>
                                                <td>{item.payee.payeeName}</td>
                                                <td>{item.status}</td>
                                            </tr>
                                        ))
                                    }
                                </tbody>
                            </table>
                        </div>
                    </div>
                </div>
            </section>
        </div>
    );
}

export default TransactionHistory;
