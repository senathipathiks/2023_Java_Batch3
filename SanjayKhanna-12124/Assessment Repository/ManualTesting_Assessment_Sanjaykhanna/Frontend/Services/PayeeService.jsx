import axios from 'axios';
import { Component } from 'react';
const origin = "http://localhost:1330";
const addPayee = "/addPayee"
const deletePayee = "/deletePayee?payeeId="
const updatePayee = "/updatePayee"
const getPayees = "/getPayees"
const getPayee = "/getPayee?payeeId="

class PayeeService extends Component {
 
    doAdd = (data) =>{
        return axios.post(origin+addPayee,data);
    }

    doDelete = (payeeId) =>{
        return axios.delete(origin+deletePayee+payeeId);
    }

    doUpdate = (data) => {
        return axios.put(origin+updatePayee,data);
    }

    doGetPayee = (payeeId) =>{
        return axios.get(origin+getPayee+payeeId);
    }

    doGetAll = () => {
        return axios.get(origin+getPayees);
    }
}

const payeeService = new PayeeService();
export default payeeService;
