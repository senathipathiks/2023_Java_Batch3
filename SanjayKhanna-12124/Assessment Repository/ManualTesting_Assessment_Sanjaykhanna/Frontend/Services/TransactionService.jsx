import axios from 'axios';
import { Component } from 'react';
const origin = "http://localhost:1330";
const saveTransaction = "/saveTransaction"
const getTransactions = "/getTransactions"
class TransactionService extends Component {
  
    doSave = (data) => {
        return axios.post(origin+saveTransaction,data);
    }

    doGetAll = () =>{
        return axios.get(origin+getTransactions);
    }
}
const transactionService = new TransactionService();
export default transactionService;
