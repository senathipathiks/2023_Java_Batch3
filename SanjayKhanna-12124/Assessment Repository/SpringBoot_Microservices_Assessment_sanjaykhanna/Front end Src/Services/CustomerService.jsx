import axios from 'axios';
import { Component } from 'react';

const origin = "http://localhost:1260/";
const createCustomer = "createcustomer";
const readAllCustomer = "readallcustomer";
const updateCustomer = "updatecustomer";
const deleteCustomer = "deletecustomer?id=";
const readCustomer = "readcustomer?id=";

class CustomerService extends Component {

    doCreate = (data) => {
        return axios.post(origin + createCustomer, data);
    };

    doReadAll = () => {
        return axios.get(origin + readAllCustomer);
    };

    doRead = (id) => {
        return axios.get(origin + readCustomer + id);
    };

    doUpdate = (data) => {
        return axios.put(origin + updateCustomer, data);
    };

    doDelete = (id) => {
        return axios.delete(origin + deleteCustomer + id)
    };
}
const customerService = new CustomerService();
export default customerService;
