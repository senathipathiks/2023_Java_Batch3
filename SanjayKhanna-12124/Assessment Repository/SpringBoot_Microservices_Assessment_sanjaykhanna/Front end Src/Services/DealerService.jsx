import axios from 'axios';
import { Component } from 'react';

const origin = "http://localhost:1260/";
const createDealer = "createdealer";
const readAllDealer = "readalldealer";
const updateDealer = "updatedealer";
const deleteDealer = "deletedealer?id=";
const readDealer = "readdealer?id=";
class DealerService extends Component {

    doCreate = (data) => {
        return axios.post(origin + createDealer, data);
    };

    doReadAll = () => {
        return axios.get(origin + readAllDealer);
    };

    doRead = (id) => {
        return axios.get(origin + readDealer + id);
    };

    doUpdate = (data) => {
        return axios.put(origin + updateDealer, data);
    };

    doDelete = (id) => {
        return axios.delete(origin + deleteDealer + id)
    };

}
const  dealerService = new DealerService();
export  default dealerService;