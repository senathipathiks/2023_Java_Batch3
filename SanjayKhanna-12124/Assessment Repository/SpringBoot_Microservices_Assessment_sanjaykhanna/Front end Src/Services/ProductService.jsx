import axios from 'axios';
import { Component } from 'react';

const origin = "http://localhost:1260/";
const createProduct = "createproduct";
const readAllProduct = "readallproduct";
const updateProduct = "updateproduct";
const deleteProduct = "deleteproduct?id=";
const readProduct = "readproduct?id=";

class ProductService extends Component {

    doCreate = (data) => {
        return axios.post(origin + createProduct, data);
    };

    doReadAll = () => {
        return axios.get(origin + readAllProduct);
    };

    doRead = (id) => {
        return axios.get(origin + readProduct + id);
    };

    doUpdate = (data) =>{
        return axios.put(origin+updateProduct, data); 
    };

    doDelete = (id) => {
        return axios.delete(origin + deleteProduct + id)
    };

}
const  productService = new ProductService();
export default productService;
