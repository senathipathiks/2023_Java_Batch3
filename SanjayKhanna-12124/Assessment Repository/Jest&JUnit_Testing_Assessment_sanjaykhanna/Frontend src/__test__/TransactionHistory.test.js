import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TransactionHistory from '../Components/MainComponents/TransactionHistory';
import { doGetAllTransaction } from '../Components/Services/Services';

describe('Checking Transaction History Page', () => {

    test('should render the Transaction History Page', () => {
        render(<TransactionHistory />)
        expect(screen.getByText("History Of Transaction")).toBeInTheDocument();
    });

    test('should check rendering the table', () => {
        render(<TransactionHistory />)
        expect(screen.getByTestId("historyTable")).toBeDefined();
    });

    test('should return history of transaction list', () => {
        const result = doGetAllTransaction();
        expect(result).toBeDefined();
        result.then(res => {
            expect(res.status).toEqual(200);
            expect(res.data).toEqual([]);
        });
    });

    test('Snapshot Testing for History Page', () => {
        expect(renderer.create(<TransactionHistory />).toJSON()).toMatchSnapshot();
    });
});
