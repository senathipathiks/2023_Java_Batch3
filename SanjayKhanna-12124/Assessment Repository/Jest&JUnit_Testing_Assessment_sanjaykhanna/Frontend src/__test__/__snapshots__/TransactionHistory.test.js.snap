// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking Transaction History Page Snapshot Testing for History Page 1`] = `
<div>
  <section
    className="container container-lg"
  >
    <div
      className="card card-lg mt-5 shadow-lg p-2 mx-auto"
    >
      <div
        className="card-header card-title border-0 text-center h1"
      >
        History Of Transaction
      </div>
      <div
        className="card-body"
      >
        <div
          className="container row mt-3"
        >
          <div
            className="col-6"
            style={
              Object {
                "textAlign": "left",
              }
            }
          />
        </div>
        <div
          className="container"
        >
          <table
            className="table table-bordered table-striped table-hover mt-3"
            data-testid="historyTable"
          >
            <thead
              className="bg-secondary text-white text-center"
            >
              <tr>
                <td>
                  Transaction Number
                </td>
                <td>
                  Transaction Date
                  <small
                    style={
                      Object {
                        "fontSize": "12px",
                        "marginLeft": "5px",
                      }
                    }
                  >
                    (DD/MM/YYYY)
                  </small>
                </td>
                <td>
                  Transaction Time
                  <small
                    style={
                      Object {
                        "fontSize": "12px",
                        "marginLeft": "5px",
                      }
                    }
                  >
                    (hh/mm/ss)
                  </small>
                </td>
                <td>
                  Transaction Amount
                </td>
                <td>
                  Account Number
                </td>
                <td>
                  Account Holder
                </td>
                <td>
                  Transaction Status
                </td>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
    </div>
  </section>
</div>
`;
